ingestion/src/metadata/ingestion/source/trino.py

```diff
@@ -102,9 +80,25 @@ def create(cls, config_dict, metadata_config_dict, ctx):
         metadata_config = MetadataServerConfig.parse_obj(metadata_config_dict)
         return cls(config, metadata_config, ctx)
 
-    def next_record(self) -> Iterable[OMetaDatabaseAndTable]:
-        inspector = inspect(self.engine)
-        if self.config.include_tables:
-            yield from self.fetch_tables(inspector, self.config.database)
-        if self.config.include_views:
-            yield from self.fetch_views(inspector, self.config.database)
+    def prepare(self):
+        self.inspector = inspect(self.engine)
+        self.schema_names = (
+            self.inspector.get_schema_names()
+            if not self.config.database
+            else [self.config.database]
+        )
+        return super().prepare()
+
+    def next_record(self) -> Iterable[Entity]:
```

Comment:
we expect OMetaDatabaseAndTable to emitted from sql sources. Why are you changing this? @amiorin

Reply:
Copy and paste issue. Fixed.